src/mc-fle2-rfds.c

```diff
@@ -241,6 +243,9 @@ mc_FLE2RangeFindDriverSpec_parse (mc_FLE2RangeFindDriverSpec_t *spec,
                                   const bson_t *in,
                                   mongocrypt_status_t *status)
 {
+   BSON_ASSERT_PARAM (in);
+   BSON_ASSERT (status || true);
```

Comment:
I see this assertion `BSON_ASSERT (status || true);` in several places in this repo. Does anyone know what it is for? It would seem to me that the `|| true` would mean that this expression would never evaluate to false, and therefor, this assertion would never be triggered (even if status is NULL). Do I read this correctly?

Reply:
This is a pattern I introduced to explicitly document a given pointer parameter may be `NULL` (as opposed to `BSON_ASSERT_PARAM (param)`) in an attempt to improve readability, as it is not immediately clear or consistent when a given pointer parameter may be `NULL`.

Reply:
So, the first instances of that `|| true` construct were introduced in 4d9009d03810c51c460c8d22067b7fed11187e4b.  I have to confess that I assumed there was a good reason and so as I rebased my work and filled in missing assertions I followed the pattern already present in the source files without putting any thought into whether it should be changed.  The commit that introduced the construct was authored by @kevinAlbs and contributed to by @eramongodb .  Perhaps one of them could provide some insight here on whether the pattern is purposeful or spurious.